### ISSUE TITLE
Support for custom metrics 

### ISSUE THREAD

**Issue by RaghavendraP** [grafana/azure-monitor-datasource#19]
Hi,

We have custom metrics declared for our VM but we are unable to see those metrics using azure datasource plugin in grafana. Are those supported? If not, will the support for them be added?

Thanks.

**Comment by daniellee** [comment]
Just checking but this is for Azure Monitor and not Application Insights?

The reason they do not show up is probably because the plugin has a whitelist of which metrics to show - not all API's support metrics yet. Can you point me to some docs about Custom Metrics for Azure Monitor or give me an example. I couldn't really find anything when I googled.

**Comment by RaghavendraP** [comment]
Hi,

I think it should come under Monitor itself instead of Application Insights. I am able to see them in metrics under Monitor in Azure portal. 

As per the docs, I am unable to get any. There are documents only to setup custom metrics for VM's (both Windows & Linux) but nothing that shows how to retrieve them by using APIs.

Below image show custom metric for one of our VM.
![image](https://user-images.githubusercontent.com/4476362/33112337-bd279c48-cf78-11e7-8f36-f30608e4ee11.png)

**Comment by kamathashwin** [comment]
metrics collected via the guest OS agent are not directly included in the Azure Monitor REST API. This is a feature we are working on where customer can opt-in to pump agent based metrics as custom metrics into Azure Monitor. after that, you can directly chart them in Grafana using the same data source.

**Comment by akkapes** [comment]
Any idea when the guest OS metrics will be available in Grafana?

**Comment by baffonso** [comment]
Hi, I am having the same problem.  Any idea when this will be fixed?

**Comment by RaghavendraP** [comment]
Hi,

Is there any update on providing support for viewing custom metrics from grafana?

**Comment by mjiderhamn** [comment]
It's a shame that metrics collected by Telegraf into Azure Monitor, as per https://docs.microsoft.com/en-us/azure/azure-monitor/platform/collect-custom-metrics-linux-telegraf , can not be accessed from Grafana

**Comment by raphaelquati** [comment]
I've implemented at #79. And I've tested internally here. We are using telegraf to send rabbitmq metrics (custom metrics) to azure.

![image](https://user-images.githubusercontent.com/815086/53993556-83e9dc80-410e-11e9-91cc-a541a1ad74d6.png)

**Comment by raphaelquati** [comment]
I've implemented at #79 
We using telegraf to send rabbitmq metrics to Azure Monitor.

![image](https://user-images.githubusercontent.com/815086/53993721-f0fd7200-410e-11e9-866f-29d7177c181b.png)

**Comment by SureshKhemka** [comment]
@raphaelquati I was just trying the same, sending metrics from Telegraph to Azure monitor, I can see the metrics in Azure Portal but cannot get the namespace in grafana azure monitor plugin. I can see that my subscription has operationinsights resoruce registered, anything else specific we need to do to get data into Grafana Azure monitor datasource?

**Comment by daniellee** [comment]
This repo is going to be archived. The Azure Monitor plugin was moved into the Grafana repository for version 6.0 of Grafana. If you are using Grafana 6.x then you will automatically start using the built-in version of the Azure Monitor datasource.

Announcement: https://grafana.com/blog/2019/02/25/grafana-v6.0-released/#azure-monitor-datasource
Documentation for Azure Monitor in Grafana: https://grafana.com/docs/features/datasources/azuremonitor/

This issue is now ported to the Grafana repo: https://github.com/grafana/grafana/issues/17220

Unfortunately there is a PR for this issue which was submitted after we had ported the plugin to Grafana. Really sorry about that - we should have closed this repository earlier.
